src/components/Blogs/Blogs.jsx

```diff
@@ -21,22 +31,40 @@ const Blogs = () => {
         (state) => state.userDetail,
     );
 
+    const followUserId = user?._id;
+
+    const { followData } = useSelector((state) => state.followData);
     useEffect(() => {
         if (isBlogError) console.log(blogMessage);
         if (isUserDetailError) console.log(userDetailMessage);
-
+        if (followUserId) dispatch(getFollowData(followUserId));
         dispatch(getAllBlogs());
         dispatch(getAllUserDetails());
 
         return () => {
+            dispatch(reset());
             dispatch(blogReset());
             dispatch(userDetailReset());
         };
-    }, [dispatch, isBlogError, blogMessage, isUserDetailError, userDetailMessage]);
+    }, [dispatch, isBlogError, blogMessage, isUserDetailError, userDetailMessage, followUserId]);
 
     const blogsData = blogs?.map((blog) => {
         const userDetail = userDetails?.find((user) => user.user === blog.user);
+        const { username, avatar, verified } = userDetail || {};
+
+        return {
+            ...blog,
+            username,
+            avatar,
+            verified,
+        };
+    });
 
+    const followingBlogData = blogs?.map((blog) => {
+        if (!followData?.following?.includes(blog.user)) {
+            return null;
+        }
+        const userDetail = userDetails?.find((user) => user.user === blog.user);
```

Comment:
```
const userDetail = userDetails?.find((user) => user.user === blog.user);
const { username, avatar, verified } = userDetail || {};

return {
    ...blog,
    username,
    avatar,
    verified,
};
```
You have this all code duplicated, so try to refactor the blogsData and followingBlogData by using external function of the duplicated of code.

Reply:
userDetailsFunction is created and duplicate codes are removed.